core/http-auth-aws/src/test/java/software/amazon/awssdk/http/auth/aws/internal/signer/DefaultAwsV4HttpSignerTest.java

```diff
@@ -231,4 +233,37 @@ public void sign_WithPayloadSigningFalseAndChunkEncodingTrueWithoutTrailer_Throw
 
         assertThrows(UnsupportedOperationException.class, () -> signer.sign(request));
     }
+
+    @Test
+    public void sign_withFlexibleChecksum_DelegatesToFlexibleChecksummer() {
+        SyncSignRequest<? extends AwsCredentialsIdentity> request = generateBasicRequest(
+            AwsCredentialsIdentity.create("access", "secret"),
+            httpRequest -> {
+            },
+            signRequest -> signRequest.putProperty(CHECKSUM_ALGORITHM, CRC32)
+        );
+
+        SyncSignedRequest signedRequest = signer.sign(request);
+
+        assertThat(signedRequest.request().firstMatchingHeader("x-amz-checksum-crc32")).isPresent();
+        assertThat(signedRequest.request().firstMatchingHeader("x-amz-content-sha256")).isPresent();
```

Comment:
Hmmm.. I'd prefer not to, since I think these tests should try to focus on the delegation part itself instead. I do think more tests should be added to your point for specific behaviors like that.